pallets/nfts/src/tests.rs

```diff
@@ -3095,131 +3067,108 @@ fn clear_collection_approvals_works() {
-				Ok(Some(WeightOf::<Test>::clear_collection_approvals(0)).into())
+				clear_collection_approvals(origin.clone(), maybe_owner, collection_id, 0),
+				Ok(Some(WeightOf::clear_collection_approvals(0)).into())
 			);
-			assert_eq!(Balances::free_balance(&owner), balance - 2);
+			assert_eq!(Balances::free_balance(&item_owner), balance - approvals);
 			assert_eq!(
-				CollectionApprovals::<Test>::iter_prefix((collection_id, owner.clone())).count(),
-				2
+				CollectionApprovals::iter_prefix((collection_id, &item_owner)).count(),
+				approvals as usize
 			);
 			assert!(!events().contains(&Event::<Test>::ApprovalsCancelled {
 				collection: collection_id,
 				item: None,
-				owner: owner.clone(),
+				owner: item_owner.clone(),
 			}));
 
 			// Partially removes collection approvals.
 			assert_eq!(
 				clear_collection_approvals(origin.clone(), maybe_owner.clone(), collection_id, 1),
-				Ok(Some(WeightOf::<Test>::clear_collection_approvals(1)).into())
+				Ok(Some(WeightOf::clear_collection_approvals(1)).into())
 			);
-			assert_eq!(Balances::free_balance(&owner), balance - 1);
+			assert_eq!(Balances::free_balance(&item_owner), balance - approvals + 1);
 			assert_eq!(
-				CollectionApprovals::<Test>::iter_prefix((collection_id, owner.clone())).count(),
-				1
-			);
-
-			// Successfully removes all collection approvals. Only charges post-dispatch weight for
-			// the removed approvals.
-			assert_eq!(
-				clear_collection_approvals(origin.clone(), maybe_owner.clone(), collection_id, 10),
-				Ok(Some(WeightOf::<Test>::clear_collection_approvals(1)).into())
+				CollectionApprovals::iter_prefix((collection_id, item_owner.clone())).count(),
+				approvals as usize - 1
 			);
 			assert!(events().contains(&Event::<Test>::ApprovalsCancelled {
 				collection: collection_id,
 				item: None,
-				owner: owner.clone(),
+				owner: item_owner.clone(),
 			}));
-			assert_eq!(Balances::free_balance(&owner), balance);
-			assert!(CollectionApprovals::<Test>::iter_prefix((collection_id, owner.clone()))
-				.count()
-				.is_zero());
 
-			// Emitting no event if zero approvals removed.
+			// Successfully remove all collection approvals. Only charges post-dispatch weight for
+			// the removed approvals.
 			assert_eq!(
 				clear_collection_approvals(origin.clone(), maybe_owner.clone(), collection_id, 10),
-				Ok(Some(WeightOf::<Test>::clear_collection_approvals(0)).into())
+				Ok(Some(WeightOf::clear_collection_approvals(9)).into())
 			);
-			assert!(!events().contains(&Event::<Test>::ApprovalsCancelled {
+			assert_eq!(Balances::free_balance(&item_owner), balance);
+			assert!(CollectionApprovals::iter_prefix((collection_id, item_owner.clone()))
+				.count()
+				.is_zero());
+			assert!(events().contains(&Event::<Test>::ApprovalsCancelled {
 				collection: collection_id,
 				item: None,
-				owner: owner.clone(),
+				owner: item_owner.clone(),
 			}));
-
-			assert_noop!(
-				Nfts::transfer(
```

Comment:
just let you know that this follows the behaviour of the `clear_all_transfer_approvals` test.

Reply:
Not necessary as the `CollectionApprovals` storage item is 0 and tested.

Reply:
Added it anyway :)